### DISCUSSION TITLE
Unpoly 3.1 released

### DISCUSSION THREAD

**Discussion by triskweline** [unpoly/unpoly#481] [Announcements]
This release addresses some issues when upgrading from Unpoly 2 to 3:

- Fix a bug where kept [`[up-keep]`](https://unpoly.com/up-keep) elements would call their destructors if the `<body>` element is swapped
- [Validation](https://unpoly.com/validation) now throw an exception if a validation target cannot be matched (fixes [#476](https://github.com/unpoly/unpoly/issues/476))
- Fix a bug where focused date inputs would trigger a validation when destroyed
- [Cache revalidation](https://unpoly.com/caching#revalidation) is now only the default when [navigating](https://unpoly.com/navigation). If you render cached content without navigating, you must opt into cache revalidation with `{ cache: 'auto', revalidate: 'auto' }`.

If also fixes some bugs in [`unpoly-migrate.js`](https://unpoly.com/changes/upgrading):

- Fix a bug where the deprecated origin shorthand (`&`) in attribute selector values that contain both square brackets and ampersands (fixes [#478](https://github.com/unpoly/unpoly/issues/478))
- Fix a bug where `up.$on()` was not polyfilled properly
- Fix a bug where `up.$off()` was not polyfilled properly
- Fix a bug where `up.$compiler()` was not polyfilled properly
- Fix a bug where `up.$macro()` was not polyfilled properly

Finally this release publishes a small feature:

- Published a new attribute `[up-href]`. Using this attribute you can make any element behave like a hyperlink when clicked.

**Comment by naveencom** [comment]
Thanks a lot.
